scripts/proto/lbann_proto.py

```diff
@@ -41,85 +42,78 @@ def _add_to_module_namespace(stuff):
     for k, v in stuff.items():
         g[k] = v
 
-class Module:
-    """Basic building block for layers (or sets of layers).
+class Layer:
+    """Base class for layers."""
 
-    Inherit from this for constructing neural networks. Modules are constructed
-    by calling an instance (e.g. module_instance(x)) to make x a parent of
-    module_instance.
+    num_layers = 0  # Static counter, used for default layer names
 
-    Internally, that will invoke the add_parent method, which can be overriden
-    by child modules to customize adding parents.
-
-    """
-
-    def __init__(self, name, data_layout):
-        self.name = name
-        self.data_layout = data_layout
-        self.parents = []
-        self.children = []
+    def __init__(self, name='', inputs=[], outputs=[],
+                 weights=[], data_layout=''):
+        Layer.num_layers += 1
+        self.name = name if name else 'layer{0}'.format(Layer.num_layers-1)
+        self.inputs = []
+        self.outputs = []
         self.weights = []
+        self.data_layout = data_layout
 
-    def __call__(self, parent):
-        """Make parent a parent of this module."""
-        self.add_parent(parent)
-        return self
-
-    def add_parent(self, parent):
-        """Make parent a parent of this module."""
-        self.parents.append(parent)
-        parent.children.append(self)
-
-    def proto(self, model):
-        """Add this module to the protobuf model."""
-        raise NotImplementedError('proto not implemented')
-
-    def add_weights(self, initializer, name_suffix=None):
-        """Create a new weights object and add it to this module."""
-        if name_suffix is None:
-            name_suffix = '_weights{0}'.format(len(self.weights) + 1)
-        w = Weights(self.name + name_suffix, initializer)
+        # Initialize inputs, outputs, and weights
+        for l in (inputs
+                  if isinstance(inputs, collections.Iterable)
```

Comment:
The check should be `isinstance(inputs, collections.abc.Iterable)`. Using the abstract base classes from `collections` is deprecated (and will be removed in Python 3.8).

Reply:
Done.